### PR TITLE
updating file from original

### PR DESCRIPTION
**Pull request by jinz32**

### CODE REVIEW

File: Weather/Weather/Model/Weather.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import Foundation
+
+struct WeatherModel: Codable {
+    let name: String
+    let wind: Wind
+    let main: Main
+    let weather: [Weather]
+    
+}
+struct Wind: Codable {
```
Comment (MugiwaraxD) [inline Weather/Weather/Model/Weather.swift]:
```suggestion

struct Wind: Codable {
```

File: Weather/Weather/Model/Weather.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+struct Main: Codable {
+    let temp: Double
+    let feels_like: Double
+    let temp_min: Double
+    let temp_max: Double
+    let pressure: Int
+    let humidity: Int
+}
+struct GroupWeatherResponses: Codable{
```
Comment (MugiwaraxD) [inline Weather/Weather/Model/Weather.swift:44]:
```suggestion

struct GroupWeatherResponses: Codable{
```

File: Weather/Weather/View/DetailView.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+//  Weather
+//
+//  Created by Jonathan Zheng on 3/19/24.
+//
+
+import SwiftUI
+
+struct DetailView: View {
+    @ObservedObject var viewModel = WeatherViewModel()
+    @State var cityName = " "
```
Comment (MugiwaraxD) [inline Weather/Weather/View/DetailView.swift]:
```suggestion
    @State var cityName = ""
    
```

File: Weather/Weather/View/DetailView.swift [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            .textFieldStyle(RoundedBorderTextFieldStyle())
+            .onSubmit{
+                viewModel.getWeather(for: cityName)
+            }
+        output()
+    }
+    
+    func kelvinToCelsius(_ kelvin: Double) -> Double {
+        return kelvin - 273.15
+    }
```
Comment (MugiwaraxD) [inline Weather/Weather/View/DetailView.swift]: there's an api param you can pass to open weather api to get celsius

File: Weather/Weather/ViewModel/WeatherViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+//
+//  Created by Jonathan Zheng on 3/19/24.
+//
+
+import Foundation
+import SwiftUI
+
+class WeatherViewModel: ObservableObject {
+    @Published var weatherData: WeatherModel?
+    @Published var iconImage: UIImage?
```
Comment (MugiwaraxD) [inline Weather/Weather/ViewModel/WeatherViewModel.swift]: this doesn't need to exist, it's apart of the weatherData

Comment (jinz32) [reply Weather/Weather/ViewModel/WeatherViewModel.swift]: WeatherData needs to exist doesnt it?

File: Weather/Weather/Model/Weather.swift (hunk 2 above)
Comment (jinz32) [reply Weather/Weather/Model/Weather.swift:44]: updated formatting

File: Weather/Weather/Model/Weather.swift (hunk 1 above)
Comment (jinz32) [reply Weather/Weather/Model/Weather.swift]: Updated

**Review by MugiwaraxD — APPROVED** [review]

**Pull request merged**